lib/adb.js

```diff
@@ -580,7 +580,8 @@ ADB.prototype.ps = function ps(deviceId, callback) {
 			callback(err);
 		} else {
 			// old ps, does not support '-A' parameter
-			if (data.toString().startsWith('bad pid \'-A\'')) {
+			var dataStr = data.toString();
+			if (dataStr.startsWith('bad pid \'-A\'') || dataStr.endsWith('NAME\r\n')) {
```

Comment:
Given that Android is Linux based, are you sure there's a `\r` character? Might be safer to do `var dataStr = data.toString().trim();` and then do the `endsWith()`.